contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,541 @@
+    _;
+  }
+
+  function initialise(uint256 _globalThreshold) public {
+    require(colony.hasUserRole(msgSender(), 1, ColonyDataTypes.ColonyRole.Root), "multisig-permissions-not-core-root");
+    globalThreshold = _globalThreshold;
+
+    emit ExtensionInitialised();
+  }
+
+  function getGlobalThreshold() public view returns (uint256) {
+    return globalThreshold;
+  }
+
+  function createMotion(uint256 _permissionDomainId, uint256 _childSkillIndex, address[] memory _targets, bytes[] memory _data) notDeprecated public {
+    require(_targets.length == _data.length, "colony-multisig-invalid-motion");
+    require(_targets.length >= 1, "colony-multisig-invalid-motion");
+
+    motionCount += 1;
+    Motion storage motion = motions[motionCount];
+    motion.targets = _targets;
+    motion.data = _data;
+
+    uint256 domainSkillId;
+    bytes32 permissions;
+
+    for (uint256 i = 0; i < motion.data.length; i += 1) {
+      uint256 actionDomainSkillId;
+      bytes32 actionRequiredPermissions;
+
+      (actionDomainSkillId, actionRequiredPermissions) = getActionSummary(motion.data[i], _targets[i]);
+
+      if (motion.domainSkillId == 0 && motion.requiredPermissions == 0){
+        motion.domainSkillId = actionDomainSkillId;
+        motion.requiredPermissions = actionRequiredPermissions;
+      } else {
+        require(motion.domainSkillId == actionDomainSkillId && motion.requiredPermissions == actionRequiredPermissions, "colony-multisig-invalid-motion");
+      }
+    }
+
+    // Get permissions in _permissionDomainId for user
+    bytes32 userPermissions = getUserRoles(msgSender(), _permissionDomainId);
+    // Work out which domain they're using their permissions in, via child skill index
+    uint256 permissionSkillId = colony.getDomain(_permissionDomainId).skillId;
+    uint256 userActingInDomainWithSkillId = colonyNetwork.getChildSkillId(permissionSkillId, _childSkillIndex);
+
+    // Check it's same as the domain being acted on by _motionId
+    require(userActingInDomainWithSkillId == motion.domainSkillId, "colony-multisig-not-same-domain");
+
+    // Check they have the permissions
+    require((userPermissions & motion.requiredPermissions) != bytes32(0), "colony-multisig-no-permissions");
+
+    emit MotionCreated(msgSender(), motionCount);
+
+    // TODO: do this per-permission
+    changeApproval(_permissionDomainId, _childSkillIndex, motionCount, true);
+  }
+
+  function changeApproval(uint256 _permissionDomainId, uint256 _childSkillIndex, uint256 _motionId, bool _approved) public notExecuted(_motionId) {
+    // Get permissions in _permissionDomainId for user
+    bytes32 userPermissions = getUserRoles(msgSender(), _permissionDomainId);
+
+    // Work out which domain they're using their permissions in, via child skill index
+    uint256 permissionSkillId = colony.getDomain(_permissionDomainId).skillId;
+    uint256 userActingInDomainWithSkillId = colonyNetwork.getChildSkillId(permissionSkillId, _childSkillIndex);
```

Comment:
I would put these three statements in a separate function, since it's re-used. It would keep this and the previous functions more compact as well:

```
function validateMotionDomain(uint256 _permissionDomainId, uint256 _childSkillIndex, uint256 _motionId) {
    uint256 permissionSkillId = colony.getDomain(_permissionDomainId).skillId;
    uint256 userActingInDomainWithSkillId = colonyNetwork.getChildSkillId(permissionSkillId, _childSkillIndex);
    require(userActingInDomainWithSkillId == motions[_motionId].domainSkillId, "colony-multisig-not-same-domain");
}
```
    
```
 function validateUserPermissions(uint256 _permissionDomainId, uint256 _motionId) {
    bytes32 userPermissions = getUserRoles(msgSender(), _permissionDomainId);
    require(userPermissions & motions[_motionId].requiredPermissions > 0, "colony-multisig-no-permissions");
}
```